In [5]:
import pandas as pd
import numpy as np
import matplotlib as plt
import json


In [6]:
movies_df = pd.read_csv('tmdb_5000_movies.csv')
movies_df.head()

budget                                             genres      id  \
0  237000000  [{"id": 28, "name": "Action"}, {"id": 12, "nam...   19995   
1  300000000  [{"id": 12, "name": "Adventure"}, {"id": 14, "...     285   
2  245000000  [{"id": 28, "name": "Action"}, {"id": 12, "nam...  206647   
3  250000000  [{"id": 28, "name": "Action"}, {"id": 80, "nam...   49026   
4  260000000  [{"id": 28, "name": "Action"}, {"id": 12, "nam...   49529   

                                            keywords original_language  \
0  [{"id": 1463, "name": "culture clash"}, {"id":...                en   
1  [{"id": 270, "name": "ocean"}, {"id": 726, "na...                en   
2  [{"id": 470, "name": "spy"}, {"id": 818, "name...                en   
3  [{"id": 849, "name": "dc comics"}, {"id": 853,...                en   
4  [{"id": 818, "name": "based on novel"}, {"id":...                en   

                                            overview  popularity  \
0  In the 22nd century, a paraplegic Marine is di...  150.437577   
1  Captain Barbossa, long believed to be dead, ha...  139.082615   
2  A cryptic message from Bond’s past sends him o...  107.376788   
3  Following the death of District Attorney Harve...  112.312950   
4  John Carter is a war-weary, former military ca...   43.926995   

                                production_companies  \
0  [{"name": "Ingenious Film Partners", "id": 289...   
1  [{"name": "Walt Disney Pictures", "id": 2}, {"...   
2  [{"name": "Columbia Pictures", "id": 5}, {"nam...   
3  [{"name": "Legendary Pictures", "id": 923}, {"...   
4        [{"name": "Walt Disney Pictures", "id": 2}]   

                                production_countries release_date     revenue  \
0  [{"iso_3166_1": "US", "name": "United States o...   10-12-2009  2787965087   
1  [{"iso_3166_1": "US", "name": "United States o...   19-05-2007   961000000   
2  [{"iso_3166_1": "GB", "name": "United Kingdom"...   26-10-2015   880674609   
3  [{"iso_3166_1": "US", "name": "United States o...   16-07-2012  1084939099   
4  [{"iso_3166_1": "US", "name": "United States o...   07-03-2012   284139100   

   runtime                                         tagline  \
0    162.0                     Enter the World of Pandora.   
1    169.0  At the end of the world, the adventure begins.   
2    148.0                           A Plan No One Escapes   
3    165.0                                 The Legend Ends   
4    132.0            Lost in our world, found in another.   

                                      title  vote_average  vote_count  
0                                    Avatar           7.2       11800  
1  Pirates of the Caribbean: At World's End           6.9        4500  
2                                   Spectre           6.3        4466  
3                     The Dark Knight Rises           7.6        9106  
4                               John Carter           6.1        2124

In [7]:
#Creating a list of dictionaries 
genre_column_list =[]
for each in movies_df['genres']:
    genre_column_list.append(json.loads(each))

In [8]:
genre_column_list
movieid_genre_list = []
for i,x in enumerate(genre_column_list):
    for y in x:
        movieid_genre_list.append(movies_df['id'][i])
movieid_genre_list[1:5]

[19995, 19995, 19995, 285]

In [9]:
genre_list = []
genre_id_list=[]
for x in genre_column_list:
    for y in x:
        genre_id_list.append(y['id'])
        genre_list.append(y['name'])
            

In [10]:
genre_list
genre_df = pd.DataFrame(genre_list, columns=['Genre'])
genre_df['ID'] = pd.Series(genre_id_list)
genre_df['Movie_ID'] = pd.Series(movieid_genre_list)
genre_df = genre_df[['Movie_ID','ID', 'Genre']]

In [11]:
genre_df

Movie_ID     ID            Genre
0         19995     28           Action
1         19995     12        Adventure
2         19995     14          Fantasy
3         19995    878  Science Fiction
4           285     12        Adventure
5           285     14          Fantasy
6           285     28           Action
7        206647     28           Action
8        206647     12        Adventure
9        206647     80            Crime
10        49026     28           Action
11        49026     80            Crime
12        49026     18            Drama
13        49026     53         Thriller
14        49529     28           Action
15        49529     12        Adventure
16        49529    878  Science Fiction
17          559     14          Fantasy
18          559     28           Action
19          559     12        Adventure
20        38757     16        Animation
21        38757  10751           Family
22        99861     28           Action
23        99861     12        Adventure
24        99861    878  Science Fiction
25          767     12        Adventure
26          767     14          Fantasy
27          767  10751           Family
28       209112     28           Action
29       209112     12        Adventure
...         ...    ...              ...
12096     86304    878  Science Fiction
12097     86304     53         Thriller
12098       692     27           Horror
12099       692     35           Comedy
12100       692     80            Crime
12101     39851     18            Drama
12102     13898     18            Drama
12103     13898  10769          Foreign
12104    157185     27           Horror
12105     36095     80            Crime
12106     36095     27           Horror
12107     36095   9648          Mystery
12108     36095     53         Thriller
12109    182291     18            Drama
12110    286939     53         Thriller
12111    286939     27           Horror
12112    286939     35           Comedy
12113    124606     18            Drama
12114     14337    878  Science Fiction
12115     14337     18            Drama
12116     14337     53         Thriller
12117      9367     28           Action
12118      9367     80            Crime
12119      9367     53         Thriller
12120     72766     35           Comedy
12121     72766  10749          Romance
12122    231617     35           Comedy
12123    231617     18            Drama
12124    231617  10749          Romance
12125    231617  10770         TV Movie

[12126 rows x 3 columns]

In [12]:
productioncomp_column_list = []
for each in movies_df['production_companies']:
    productioncomp_column_list.append(json.loads(each))
productioncomp_column_list

[[{'id': 289, 'name': 'Ingenious Film Partners'},
  {'id': 306, 'name': 'Twentieth Century Fox Film Corporation'},
  {'id': 444, 'name': 'Dune Entertainment'},
  {'id': 574, 'name': 'Lightstorm Entertainment'}],
 [{'id': 2, 'name': 'Walt Disney Pictures'},
  {'id': 130, 'name': 'Jerry Bruckheimer Films'},
  {'id': 19936, 'name': 'Second Mate Productions'}],
 [{'id': 5, 'name': 'Columbia Pictures'},
  {'id': 10761, 'name': 'Danjaq'},
  {'id': 69434, 'name': 'B24'}],
 [{'id': 923, 'name': 'Legendary Pictures'},
  {'id': 6194, 'name': 'Warner Bros.'},
  {'id': 9993, 'name': 'DC Entertainment'},
  {'id': 9996, 'name': 'Syncopy'}],
 [{'id': 2, 'name': 'Walt Disney Pictures'}],
 [{'id': 5, 'name': 'Columbia Pictures'},
  {'id': 326, 'name': 'Laura Ziskin Productions'},
  {'id': 19551, 'name': 'Marvel Enterprises'}],
 [{'id': 2, 'name': 'Walt Disney Pictures'},
  {'id': 6125, 'name': 'Walt Disney Animation Studios'}],
 [{'id': 420, 'name': 'Marvel Studios'},
  {'id': 15357, 'name': 'Prime Foc

In [13]:
company_id_list = []
company_list=[]
for x in productioncomp_column_list:
    for y in x:
        company_id_list.append(y['id'])
        company_list.append(y['name'])

In [14]:
company_id_list

[289,
 306,
 444,
 574,
 2,
 130,
 19936,
 5,
 10761,
 69434,
 923,
 6194,
 9993,
 9996,
 2,
 5,
 326,
 19551,
 2,
 6125,
 420,
 15357,
 76043,
 6194,
 7364,
 429,
 507,
 6194,
 9993,
 9995,
 41624,
 429,
 923,
 6194,
 9168,
 7576,
 2,
 130,
 19936,
 2,
 130,
 2691,
 37380,
 37381,
 37382,
 923,
 6194,
 9993,
 9996,
 78685,
 5888,
 10221,
 11345,
 11440,
 11441,
 11442,
 76043,
 4,
 420,
 2,
 130,
 20478,
 56,
 5627,
 6736,
 9169,
 11084,
 11,
 12,
 174,
 7413,
 8411,
 5,
 326,
 7505,
 23,
 33,
 1645,
 7295,
 11,
 12,
 174,
 8411,
 12,
 289,
 1473,
 1938,
 11,
 33,
 68,
 69,
 4,
 306,
 574,
 264,
 420,
 3036,
 84424,
 84425,
 33,
 2598,
 13778,
 20153,
 13,
 56,
 923,
 3341,
 6452,
 5,
 5,
 326,
 19551,
 420,
 2,
 598,
 8601,
 16314,
 20004,
 289,
 306,
 431,
 444,
 445,
 9168,
 12247,
 19551,
 79028,
 79029,
 79030,
 2,
 3,
 4,
 27,
 56,
 435,
 2481,
 22826,
 76043,
 4,
 56,
 435,
 2481,
 11413,
 22826,
 38833,
 76043,
 2,
 16314,
 5,
 19551,
 31828,
 53462,
 2,
 7161,
 18713,
 23791,

In [15]:
company_list

['Ingenious Film Partners',
 'Twentieth Century Fox Film Corporation',
 'Dune Entertainment',
 'Lightstorm Entertainment',
 'Walt Disney Pictures',
 'Jerry Bruckheimer Films',
 'Second Mate Productions',
 'Columbia Pictures',
 'Danjaq',
 'B24',
 'Legendary Pictures',
 'Warner Bros.',
 'DC Entertainment',
 'Syncopy',
 'Walt Disney Pictures',
 'Columbia Pictures',
 'Laura Ziskin Productions',
 'Marvel Enterprises',
 'Walt Disney Pictures',
 'Walt Disney Animation Studios',
 'Marvel Studios',
 'Prime Focus',
 'Revolution Sun Studios',
 'Warner Bros.',
 'Heyday Films',
 'DC Comics',
 'Atlas Entertainment',
 'Warner Bros.',
 'DC Entertainment',
 'Cruel & Unusual Films',
 'RatPac-Dune Entertainment',
 'DC Comics',
 'Legendary Pictures',
 'Warner Bros.',
 'Bad Hat Harry Productions',
 'Eon Productions',
 'Walt Disney Pictures',
 'Jerry Bruckheimer Films',
 'Second Mate Productions',
 'Walt Disney Pictures',
 'Jerry Bruckheimer Films',
 'Infinitum Nihil',
 'Silver Bullet Productions (II)',
 'B

In [16]:
movieid_prodcomp_list = []
for i, x in enumerate(productioncomp_column_list):
    for y in x:
        movieid_prodcomp_list.append(movies_df['id'][i])
movieid_prodcomp_list

[19995,
 19995,
 19995,
 19995,
 285,
 285,
 285,
 206647,
 206647,
 206647,
 49026,
 49026,
 49026,
 49026,
 49529,
 559,
 559,
 559,
 38757,
 38757,
 99861,
 99861,
 99861,
 767,
 767,
 209112,
 209112,
 209112,
 209112,
 209112,
 209112,
 1452,
 1452,
 1452,
 1452,
 10764,
 58,
 58,
 58,
 57201,
 57201,
 57201,
 57201,
 57201,
 57201,
 49521,
 49521,
 49521,
 49521,
 49521,
 2454,
 2454,
 2454,
 2454,
 2454,
 2454,
 2454,
 24428,
 24428,
 1865,
 1865,
 1865,
 41154,
 41154,
 41154,
 41154,
 41154,
 122917,
 122917,
 122917,
 122917,
 122917,
 1930,
 1930,
 1930,
 20662,
 20662,
 20662,
 20662,
 57158,
 57158,
 57158,
 57158,
 2268,
 2268,
 2268,
 2268,
 254,
 254,
 254,
 254,
 597,
 597,
 597,
 271110,
 271110,
 271110,
 271110,
 271110,
 44833,
 44833,
 44833,
 44833,
 135397,
 135397,
 135397,
 135397,
 135397,
 37724,
 558,
 558,
 558,
 68721,
 12155,
 12155,
 12155,
 12155,
 12155,
 36668,
 36668,
 36668,
 36668,
 36668,
 36668,
 36668,
 36668,
 36668,
 36668,
 36668,
 62211,
 6

In [17]:
prod_companies_df = pd.DataFrame(movieid_prodcomp_list, columns = ['Movie_ID'])
prod_companies_df['ID'] = pd.Series(company_id_list)
prod_companies_df['Company'] = pd.Series(company_list)
prod_companies_df

Movie_ID     ID                                 Company
0         19995    289                 Ingenious Film Partners
1         19995    306  Twentieth Century Fox Film Corporation
2         19995    444                      Dune Entertainment
3         19995    574                Lightstorm Entertainment
4           285      2                    Walt Disney Pictures
5           285    130                 Jerry Bruckheimer Films
6           285  19936                 Second Mate Productions
7        206647      5                       Columbia Pictures
8        206647  10761                                  Danjaq
9        206647  69434                                     B24
10        49026    923                      Legendary Pictures
11        49026   6194                            Warner Bros.
12        49026   9993                        DC Entertainment
13        49026   9996                                 Syncopy
14        49529      2                    Walt Disney Pictures
15          559      5                       Columbia Pictures
16          559    326                Laura Ziskin Productions
17          559  19551                      Marvel Enterprises
18        38757      2                    Walt Disney Pictures
19        38757   6125           Walt Disney Animation Studios
20        99861    420                          Marvel Studios
21        99861  15357                             Prime Focus
22        99861  76043                  Revolution Sun Studios
23          767   6194                            Warner Bros.
24          767   7364                            Heyday Films
25       209112    429                               DC Comics
26       209112    507                     Atlas Entertainment
27       209112   6194                            Warner Bros.
28       209112   9993                        DC Entertainment
29       209112   9995                   Cruel & Unusual Films
...         ...    ...                                     ...
13618      2292     14                           Miramax Films
13619      2292  16934                  View Askew Productions
13620     42497   3923                        Strand Releasing
13621     14585    803        Alliance Atlantis Communications
13622     14585   8809             Fair and Square Productions
13623     38780   1062    Boll Kino Beteiligungs GmbH & Co. KG
13624     38780   1063                    Brightlight Pictures
13625     38780   5723                Pitchblack Pictures Inc.
13626     38780  22793                 Event Film Distribution
13627     38780  22794                        Amok Productions
13628    226458   2943                          GO Productions
13629    226458  35591                 Vicarious Entertainment
13630    226458  52909                   Bloodline Productions
13631    226458  52910                           Shallow Focus
13632     86304  18818                               Grindfest
13633       692    407                   Dreamland Productions
13634     13898   2835           Jafar Panahi Film Productions
13635    157185  21871                              Park Films
13636    157185  79821                      Camera Stylo Films
13637     36095    881                           Daiei Studios
13638    182291  26677                        Iconoclast Films
13639    286939  37870                         Gold Lion Films
13640    286939  37871                    T-Street Productions
13641    124606  10571                            Asylum Films
13642    124606  26598                        FM Entertainment
13643    124606  40739                    Eagle Eye Films Inc.
13644     14337    446                               Thinkfilm
13645      9367      5                       Columbia Pictures
13646    231617   3958                   Front Street Pictures
13647    231617   6438          Muse Entertainment Enterprises

[13648 rows x 3 columns]

In [18]:
cast_xl_df = pd.read_csv('tmdb_5000_cast.csv')
cast_xl_df.head()

movie_id                                     title  \
0     19995                                    Avatar   
1       285  Pirates of the Caribbean: At World's End   
2    206647                                   Spectre   
3     49026                     The Dark Knight Rises   
4     49529                               John Carter   

                                                cast  
0  [{"cast_id": 242, "character": "Jake Sully", "...  
1  [{"cast_id": 4, "character": "Captain Jack Spa...  
2  [{"cast_id": 1, "character": "James Bond", "cr...  
3  [{"cast_id": 2, "character": "Bruce Wayne / Ba...  
4  [{"cast_id": 5, "character": "John Carter", "c...

In [27]:
cast_column_list =[]
for each in cast_xl_df['cast'][:218]:
    cast_column_list.append(json.loads(each))

In [28]:
for each in cast_xl_df['cast'][219:598]:
    cast_column_list.append(json.loads(each))

In [29]:
for each in cast_xl_df['cast'][599:637]:
    cast_column_list.append(json.loads(each))

In [30]:
for each in cast_xl_df['cast'][638:4208]:
    cast_column_list.append(json.loads(each))

In [31]:
for each in cast_xl_df['cast'][4209:]:
    cast_column_list.append(json.loads(each))

In [32]:
cast_column_list[1:10]

[[{'cast_id': 4,
   'character': 'Captain Jack Sparrow',
   'credit_id': '52fe4232c3a36847f800b50d',
   'gender': 2,
   'id': 85,
   'name': 'Johnny Depp',
   'order': 0},
  {'cast_id': 5,
   'character': 'Will Turner',
   'credit_id': '52fe4232c3a36847f800b511',
   'gender': 2,
   'id': 114,
   'name': 'Orlando Bloom',
   'order': 1},
  {'cast_id': 6,
   'character': 'Elizabeth Swann',
   'credit_id': '52fe4232c3a36847f800b515',
   'gender': 1,
   'id': 116,
   'name': 'Keira Knightley',
   'order': 2},
  {'cast_id': 12,
   'character': 'William "Bootstrap Bill" Turner',
   'credit_id': '52fe4232c3a36847f800b52d',
   'gender': 2,
   'id': 1640,
   'name': 'Stellan Skarsgård',
   'order': 3},
  {'cast_id': 10,
   'character': 'Captain Sao Feng',
   'credit_id': '52fe4232c3a36847f800b525',
   'gender': 2,
   'id': 1619,
   'name': 'Chow Yun-fat',
   'order': 4},
  {'cast_id': 9,
   'character': 'Captain Davy Jones',
   'credit_id': '52fe4232c3a36847f800b521',
   'gender': 2,
   'id': 24

In [33]:
final_cast_column =[]
for x in cast_column_list:
    dummy_column=[]
    for y in x[:5]:
        dummy_column.append(y)
    final_cast_column.append(dummy_column)
final_cast_column

[[{'cast_id': 242,
   'character': 'Jake Sully',
   'credit_id': '5602a8a7c3a3685532001c9a',
   'gender': 2,
   'id': 65731,
   'name': 'Sam Worthington',
   'order': 0},
  {'cast_id': 3,
   'character': 'Neytiri',
   'credit_id': '52fe48009251416c750ac9cb',
   'gender': 1,
   'id': 8691,
   'name': 'Zoe Saldana',
   'order': 1},
  {'cast_id': 25,
   'character': 'Dr. Grace Augustine',
   'credit_id': '52fe48009251416c750aca39',
   'gender': 1,
   'id': 10205,
   'name': 'Sigourney Weaver',
   'order': 2},
  {'cast_id': 4,
   'character': 'Col. Quaritch',
   'credit_id': '52fe48009251416c750ac9cf',
   'gender': 2,
   'id': 32747,
   'name': 'Stephen Lang',
   'order': 3},
  {'cast_id': 5,
   'character': 'Trudy Chacon',
   'credit_id': '52fe48009251416c750ac9d3',
   'gender': 1,
   'id': 17647,
   'name': 'Michelle Rodriguez',
   'order': 4}],
 [{'cast_id': 4,
   'character': 'Captain Jack Sparrow',
   'credit_id': '52fe4232c3a36847f800b50d',
   'gender': 2,
   'id': 85,
   'name': 'Jo

In [34]:
movieid_list = list(cast_xl_df['movie_id'])
for i in sorted([218,598,637,4208], reverse=True):
    del movieid_list[i]
len(movieid_list)

4757

In [35]:
movieid_cast_list=[]
for i, x in enumerate(final_cast_column):
    for y in x:
        movieid_cast_list.append(movieid_list[i])
len(movieid_cast_list)

23564

In [36]:
cast_id_list = []
cast_charname_list = []
cast_name_list =[]
for y in final_cast_column:
    for x in y:
        cast_id_list.append(x['id'])
        cast_charname_list.append(x['character'])
        cast_name_list.append(x['name'])

In [37]:
cast_df = pd.DataFrame(movieid_cast_list, columns=['Movie_ID'])
cast_df['ID'] = pd.Series(cast_id_list)
cast_df['Character'] = pd.Series(cast_charname_list)
cast_df['Actor'] = pd.Series(cast_name_list)
cast_df

Movie_ID       ID                        Character  \
0         19995    65731                       Jake Sully   
1         19995     8691                          Neytiri   
2         19995    10205              Dr. Grace Augustine   
3         19995    32747                    Col. Quaritch   
4         19995    17647                     Trudy Chacon   
5           285       85             Captain Jack Sparrow   
6           285      114                      Will Turner   
7           285      116                  Elizabeth Swann   
8           285     1640  William "Bootstrap Bill" Turner   
9           285     1619                 Captain Sao Feng   
10       206647     8784                       James Bond   
11       206647    27319                          Blofeld   
12       206647   121529                        Madeleine   
13       206647     5469                                M   
14       206647    28782                            Lucia   
15        49026     3894             Bruce Wayne / Batman   
16        49026     3895                Alfred Pennyworth   
17        49026       64                     James Gordon   
18        49026     1813           Selina Kyle / Catwoman   
19        49026     2524                             Bane   
20        49529    60900                      John Carter   
21        49529    21044                     Dejah Thoris   
22        49529     2206                             Sola   
23        49529     5293                      Tars Tarkas   
24        49529    19159                        Tal Hajus   
25          559     2219        Peter Parker / Spider-Man   
26          559      205                 Mary Jane Watson   
27          559    17051        Harry Osborn / New Goblin   
28          559    19159            Flint Marko / Sandman   
29          559    17052              Eddie Brock / Venom   
...         ...      ...                              ...   
23534     14337    76624                            Aaron   
23535     14337    76625                              Abe   
23536     14337   110171                           Robert   
23537     14337   110172                           Philip   
23538     14337   110173                             Kara   
23539      9367    20498                      El Mariachi   
23540      9367    57470                          Bigotón   
23541      9367    57471                  Mauricio (Moco)   
23542      9367  1192733                             Azul   
23543      9367  1192734                        Cantinero   
23544     72766    12833                            Buzzy   
23545     72766    58356                            Linda   
23546     72766    17188                           Marsha   
23547     72766   211993                            Katie   
23548     72766  1257819                          Vanessa   
23549    231617    11867                   Oliver O’Toole   
23550    231617    34915                  Shane McInerney   
23551    231617    58393                     Rita Haywith   
23552    231617   119810                    Norman Dorman   
23553    231617    77865                    Charlie Riggs   
23554    126186    82093                              Sam   
23555    126186   210050                           Amanda   
23556    126186     2053                           Donald   
23557    126186     2394                     Marcus Groff   
23558    126186  1395490                                    
23559     25975    69597                          Herself   
23560     25975    85563                          Himself   
23561     25975     3034                          Himself   
23562     25975    21315                          Himself   
23563     25975     2171                          Himself   

                        Actor  
0             Sam Worthington  
1                 Zoe Saldana  
2            Sigourney Weaver  
3                Stephen Lang  
4          Michelle Rodriguez  
5                 Johnny Depp  


In [38]:
movies_df=movies_df[['id','title','budget','revenue','original_language','tagline','overview','release_date','runtime','popularity','vote_average','vote_count']]
movies_df

id                                        title     budget  \
0      19995                                       Avatar  237000000   
1        285     Pirates of the Caribbean: At World's End  300000000   
2     206647                                      Spectre  245000000   
3      49026                        The Dark Knight Rises  250000000   
4      49529                                  John Carter  260000000   
5        559                                 Spider-Man 3  258000000   
6      38757                                      Tangled  260000000   
7      99861                      Avengers: Age of Ultron  280000000   
8        767       Harry Potter and the Half-Blood Prince  250000000   
9     209112           Batman v Superman: Dawn of Justice  250000000   
10      1452                             Superman Returns  270000000   
11     10764                            Quantum of Solace  200000000   
12        58   Pirates of the Caribbean: Dead Man's Chest  200000000   
13     57201                              The Lone Ranger  255000000   
14     49521                                 Man of Steel  225000000   
15      2454     The Chronicles of Narnia: Prince Caspian  225000000   
16     24428                                 The Avengers  220000000   
17      1865  Pirates of the Caribbean: On Stranger Tides  380000000   
18     41154                               Men in Black 3  225000000   
19    122917    The Hobbit: The Battle of the Five Armies  250000000   
20      1930                       The Amazing Spider-Man  215000000   
21     20662                                   Robin Hood  200000000   
22     57158          The Hobbit: The Desolation of Smaug  250000000   
23      2268                           The Golden Compass  180000000   
24       254                                    King Kong  207000000   
25       597                                      Titanic  200000000   
26    271110                   Captain America: Civil War  250000000   
27     44833                                   Battleship  209000000   
28    135397                               Jurassic World  150000000   
29     37724                                      Skyfall  200000000   
...      ...                                          ...        ...   
4744   40769                          Mutual Appreciation     500000   
4745  220490            Her Cry: La Llorona Investigation          0   
4746   42151                                 Down Terrace      31192   
4747    2292                                       Clerks      27000   
4748   42497                               Pink Narcissus      27000   
4749   33693                                  Funny Ha Ha          0   
4750   14585                        In the Company of Men          0   
4751  185465                                       Manito          0   
4752   38780                                      Rampage          0   
4753   14022                                      Slacker          0   
4754  366967                                  Dutch Kills          0   
4755  255266                                    Dry Spell      22000   
4756   17345                                     Flywheel          0   
4757  226458                                     Backmask          0   
4758   24055                              The Puffy Chair          0   
4759  287625                         Stories of Our Lives          0   
4760   44990                             Breaking Upwards          0   
4761   86304                     All Superheroes Must Die          0   
4762     692                               Pink Flamingos      12000   
4763   39851                                        Clean          0   
4764   13898                                   The Circle          0   
4765  157185                                  Tin Can Man         13   
4766   36095                                         Cure      20000   
4767  182291                               On The 

In [39]:
movies_xl_df=pd.read_csv('tmdb_5000_movies.csv')


countries_column_list =[]
for each in movies_xl_df['production_countries']:
    countries_column_list.append(json.loads(each))
    
    
movieid_countries_list = []
for i,x in enumerate(countries_column_list):
    for y in x:
        movieid_countries_list.append(movies_xl_df['id'][i])

countries_name_list=[]
for x in countries_column_list:
    for y in x:
        countries_name_list.append(y['name'])

countries_df = pd.DataFrame(countries_name_list, columns=['Shoot_Location'])
countries_df['Movie_ID'] = pd.Series(movieid_countries_list)
countries_df = countries_df[['Movie_ID', 'Shoot_Location']]
countries_df

Movie_ID            Shoot_Location
0        19995  United States of America
1        19995            United Kingdom
2          285  United States of America
3       206647            United Kingdom
4       206647  United States of America
5        49026  United States of America
6        49529  United States of America
7          559  United States of America
8        38757  United States of America
9        99861  United States of America
10         767            United Kingdom
11         767  United States of America
12      209112  United States of America
13        1452  United States of America
14       10764            United Kingdom
15       10764  United States of America
16          58                   Jamaica
17          58  United States of America
18          58                   Bahamas
19          58                  Dominica
20       57201  United States of America
21       49521            United Kingdom
22       49521  United States of America
23        2454            Czech Republic
24        2454                    Poland
25        2454                  Slovenia
26        2454  United States of America
27       24428  United States of America
28        1865  United States of America
29       41154  United States of America
...        ...                       ...
6385    290391  United States of America
6386     42151            United Kingdom
6387      2292  United States of America
6388     42497  United States of America
6389     33693  United States of America
6390     14585                    Canada
6391     14585  United States of America
6392    185465  United States of America
6393     38780                    Canada
6394     38780                   Germany
6395     14022  United States of America
6396    255266  United States of America
6397     17345  United States of America
6398    226458  United States of America
6399    287625                     Kenya
6400     44990  United States of America
6401       692  United States of America
6402     39851            United Kingdom
6403     39851                    France
6404     39851                    Canada
6405     13898                      Iran
6406    157185                   Ireland
6407     36095                     Japan
6408    182291  United States of America
6409    286939  United States of America
6410    124606  United States of America
6411     14337  United States of America
6412      9367                    Mexico
6413      9367  United States of America
6414    231617  United States of America

[6415 rows x 2 columns]

In [40]:
movieid_abv_7 = movies_df[movies_df['vote_average'] >= 7]['id']
movieid_abv_7 = list(movieid_abv_7)
movieid_abv_7

[19995,
 49026,
 38757,
 99861,
 767,
 58,
 24428,
 122917,
 57158,
 597,
 271110,
 62211,
 10193,
 168259,
 127585,
 54138,
 64682,
 10681,
 155,
 14160,
 1726,
 44826,
 137113,
 150540,
 102651,
 119450,
 100402,
 177572,
 82690,
 10191,
 118340,
 157336,
 27205,
 49051,
 4922,
 49538,
 675,
 674,
 2062,
 272,
 109445,
 76341,
 177677,
 81188,
 616,
 13475,
 82702,
 13183,
 281957,
 2503,
 102899,
 101299,
 673,
 671,
 22,
 51497,
 58574,
 861,
 87827,
 1895,
 585,
 49047,
 2567,
 120,
 286217,
 98,
 180,
 672,
 36557,
 280,
 4982,
 68718,
 11688,
 1372,
 106646,
 9016,
 29193,
 3131,
 76758,
 855,
 18,
 10501,
 12,
 122,
 121,
 61791,
 9806,
 863,
 116745,
 1422,
 10674,
 65754,
 10528,
 686,
 921,
 8358,
 2502,
 161,
 853,
 1904,
 16858,
 22538,
 1949,
 11544,
 11324,
 4147,
 228150,
 9023,
 161795,
 49948,
 36970,
 453,
 8587,
 309809,
 93456,
 9693,
 75656,
 594,
 203801,
 197,
 16869,
 587,
 57212,
 2756,
 2270,
 978,
 20352,
 9008,
 857,
 425,
 603,
 568,
 82695,
 227306,
 345,

In [41]:
actors_abv_7 = []
for i, x in enumerate(cast_df['Movie_ID']):
    if x in movieid_abv_7:
        actors_abv_7.append(cast_df['Actor'][i])
actors_abv_7 = list(set(actors_abv_7))
indices = [55,122,116,117,118]
for i in sorted(indices, reverse=True):
    del actors_abv_7[i]
len(actors_abv_7)

3096

# Scraping Twitter data using the Python-Twitter API 

In [69]:
import urllib
import oauth2 as oauth

consumer_key = 'zYBUuTWO3nnm8Rmi5HLKKANV8'
consumer_secret = 'kifuFCVkDXthnHwbEni92pKfowa32VTF1l3C6cliwfqFFue9qw'

request_token_url = 'https://api.twitter.com/oauth/request_token'
access_token_url = 'https://api.twitter.com/oauth/access_token'
authorize_url = 'https://api.twitter.com/oauth/authorize'

consumer = oauth.Consumer(consumer_key, consumer_secret)
client = oauth.Client(consumer)

# Step 1: Get a request token. This is a temporary token that is used for 
# having the user authorize an access token and to sign the request to obtain 
# said access token.
    
resp, content = client.request(request_token_url, "GET")
if resp['status'] != '200':
    raise Exception("Invalid response %s." % resp['status'])
request_token = dict(urllib.parse.parse_qsl(content))
print(request_token)
# print(request_token[b'oauth_token'])

print("Request Token:")
print("    - oauth_token        = %s" % request_token[b'oauth_token'])
print("    - oauth_token_secret = %s" % request_token[b'oauth_token_secret'])
print 

# Step 2: Redirect to the provider. Since this is a CLI script we do not 
# redirect. In a web application you would redirect the user to the URL
# below.

print("Go to the following link in your browser:")
print("%s?oauth_token=%s" % (authorize_url, request_token[b'oauth_token'].decode('ascii')))
print 

# After the user has granted access to you, the consumer, the provider will
# redirect you to whatever URL you have told them to redirect to. You can 
# usually define this in the oauth_callback argument as well.
accepted = 'n'
while accepted.lower() == 'n':
    accepted = input('Have you authorized me? (y/n) ')
oauth_verifier = input('What is the PIN? ')

# Step 3: Once the consumer has redirected the user back to the oauth_callback
# URL you can request the access token the user has approved. You use the 
# request token to sign this request. After this is done you throw away the
# request token and use the access token returned. You should store this 
# access token somewhere safe, like a database, for future use.
token = oauth.Token(request_token[b'oauth_token'],
    request_token[b'oauth_token_secret'])
token.set_verifier(oauth_verifier)
client = oauth.Client(consumer, token)

resp, content = client.request(access_token_url, "POST")
access_token = dict(urllib.parse.parse_qsl(content))
print(access_token)
print ("Access Token:")
print ("    - oauth_token        = %s" % access_token[b'oauth_token'].decode('ascii'))
print ("    - oauth_token_secret = %s" % access_token[b'oauth_token_secret'].decode('ascii'))
print
print ("You may now access protected resources using the access tokens above." )
print

{b'oauth_token': b'Nwp63QAAAAAA4nyUAAABYd5adQg', b'oauth_token_secret': b'dA4jCwFUj7lgL63BVQEjXCTHkc3zfoXk', b'oauth_callback_confirmed': b'true'}
Request Token:
    - oauth_token        = b'Nwp63QAAAAAA4nyUAAABYd5adQg'
    - oauth_token_secret = b'dA4jCwFUj7lgL63BVQEjXCTHkc3zfoXk'
Go to the following link in your browser:
https://api.twitter.com/oauth/authorize?oauth_token=Nwp63QAAAAAA4nyUAAABYd5adQg
Have you authorized me? (y/n) y
What is the PIN? 5241330
{b'oauth_token': b'848758912085934080-vtGQJbeWrgz6LzjuVLjGlEPJoqkEIld', b'oauth_token_secret': b'MUrGvzV4uR6yjzkSLYZK2yUHHNKEjPc92vzr3JNSdgqta', b'user_id': b'848758912085934080', b'screen_name': b'ibands77', b'x_auth_expires': b'0'}
Access Token:
    - oauth_token        = 848758912085934080-vtGQJbeWrgz6LzjuVLjGlEPJoqkEIld
    - oauth_token_secret = MUrGvzV4uR6yjzkSLYZK2yUHHNKEjPc92vzr3JNSdgqta
You may now access protected resources using the access tokens above.


<function print>

In [70]:
import twitter
import json

consumer_key = "F1S2szr2vhNcpJCLw3zOYkyJS"
consumer_secret = "8hkEIusC1berMbidcRlxS7AfiB5CSb1BS13y5ANfARDENfWy8k"
access_token = 	"750192775165976576-GVbo7VBvoYKYAa3GCbgpMWhFKZyoDSD"
access_token_secret = "h5SaruULKXdysSTJ93k96Xl41y1VbqOE48GG5FynNkaHv"

api = twitter.Api(consumer_key=consumer_key,
                  consumer_secret=consumer_secret,
                  access_token_key=access_token,
                  access_token_secret=access_token_secret)

In [45]:
actor_tags=[]
users=[]
# Twitter allows only 180 tweet searches per 15 minutes, so chunking is done
for each in actors_abv_7[:180]:
    each = each.replace("'",' ')
    first_and_last_name = each.split(" ")    # Splitting each actors name into first, middle and last name
    first_name = first_and_last_name[0]
    last_name=''
    middle_name=''
    if len(first_and_last_name) == 2:
        last_name = first_and_last_name[1]
    if len(first_and_last_name) == 3:
        middle_name = first_and_last_name[1]
        last_name = first_and_last_name[2]
    
    results = api.GetSearch(raw_query="q=" +first_name+"%20"+middle_name+"%20"+last_name+ "&count=100")
    cnt = len(results)
    #print(cnt)              #Making a call for each actor on twitter
    for res in results:
        x = json.loads(str(res))
        actor_info={
            'name' : first_name + ' ' + last_name,
            'tags' : [],
            'count' : cnt
        }
        user_info={
            'id': '',
            'name':''
        }
        user_info['id'] = x['user']['id']
        user_info['name'] = x['user']['name']
        users.append(user_info)
        for item in x['hashtags']:
            actor_info['tags'].append(item['text'])
    actor_tags.append(actor_info)

KeyError: 'name'

In [46]:
users = users[:500]
users

[{'id': 3146103351, 'name': 'maizoomer loves maryoomer'},
 {'id': 21213366, 'name': 'Josh Nelson'},
 {'id': 622632387, 'name': 'Pan de canela ♣️'},
 {'id': 998488676, 'name': 'Josh'},
 {'id': 3566919375, 'name': 'helena˘͈ᵕ˘͈'},
 {'id': 2921785101, 'name': 'Oliva #2'},
 {'id': 819972136135774210, 'name': 'Josh Burman'},
 {'id': 2984975175, 'name': 'tu mama en tanga'},
 {'id': 4851642419, 'name': 'gaby'},
 {'id': 1661137050, 'name': '˃̵'},
 {'id': 4851642419, 'name': 'gaby'},
 {'id': 480544088, 'name': 'Lucas Montaño'},
 {'id': 489881009, 'name': 'Derrick Mitchell'},
 {'id': 4087570773, 'name': 'Lucas'},
 {'id': 21213366, 'name': 'Josh Nelson'},
 {'id': 21213366, 'name': 'Josh Nelson'},
 {'id': 42377905, 'name': 'Jayde Lovell'},
 {'id': 54990572, 'name': 'Lucas Souza'},
 {'id': 4368379577, 'name': 'Lucas Sarubbio'},
 {'id': 934123109841559552, 'name': 'Lucas Franzotti'},
 {'id': 743771665, 'name': 'Lucas#TLoU2👾TheLastJedi'},
 {'id': 828443236960174080, 'name': 'Dan Trombley'},
 {'id': 95

In [47]:
for each in actors_abv_7[180:360]:    #Chunking 180 tweets per call as per rate limit
    each = each.replace("'",' ')
    first_and_last_name = each.split(" ")    # Splitting each actors name into first, middle and last name
    first_name = first_and_last_name[0]
    last_name=''
    middle_name=''
    if len(first_and_last_name) == 2:
        last_name = first_and_last_name[1]
    if len(first_and_last_name) == 3:
        middle_name = first_and_last_name[1]
        last_name = first_and_last_name[2]
    
    results = api.GetSearch(raw_query = "q=" + first_name + "%20"+ middle_name + "%20" + last_name + "&count=100")
    cnt = len(results)
    #print(cnt)              #Making a call for each actor on twitter
    for res in results:
        x = json.loads(str(res))
        actor_info={
            'name' : first_name + ' ' + last_name,
            'tags' : [],
            'count' : cnt
        }

        for item in x['hashtags']:
            actor_info['tags'].append(item['text'])
    actor_tags.append(actor_info)

In [44]:
for each in actors_abv_7[360:540]:
    each = each.replace("'",' ')
    first_and_last_name = each.split(" ")    #Splitting each actors name into first, middle and last name
    first_name = first_and_last_name[0]
    last_name=''
    middle_name=''
    if len(first_and_last_name) == 2:
        last_name = first_and_last_name[1]
    if len(first_and_last_name) == 3:
        middle_name = first_and_last_name[1]
        last_name = first_and_last_name[2]
    
    results = api.GetSearch(raw_query="q=" +first_name+"%20"+middle_name+"%20"+last_name + "&count=100")
    cnt = len(results)
    #print(cnt)              #Making a call for each actor on twitter
    for res in results:
        x = json.loads(str(res))
        actor_info={
            'name' : first_name + ' ' + last_name,
            'tags' : [],
            'count' : cnt
        }
        
        for item in x['hashtags']:
            actor_info['tags'].append(item['text'])
    actor_tags.append(actor_info)

In [51]:
for each in actors_abv_7[540:720]:
    each = each.replace("'",' ')
    first_and_last_name = each.split(" ")    #Splitting each actors name into first, middle and last name
    first_name = first_and_last_name[0]
    last_name=''
    middle_name=''
    if len(first_and_last_name) == 2:
        last_name = first_and_last_name[1]
    if len(first_and_last_name) == 3:
        middle_name = first_and_last_name[1]
        last_name = first_and_last_name[2]
    
    results = api.GetSearch(raw_query="q=" +first_name+"%20"+middle_name+"%20"+last_name + "&count=100")
    cnt = len(results)              #Making a call for each actor on twitter
    for res in results:
        x = json.loads(str(res))
        actor_info={
            'name' : first_name + ' ' + last_name,
            'tags' : [],
            'count' : cnt
        }
        
        for item in x['hashtags']:
            actor_info['tags'].append(item['text'])
    actor_tags.append(actor_info)

In [53]:
for each in actors_abv_7[720:900]:
    each = each.replace("'",' ')
    first_and_last_name = each.split(" ")    #Splitting each actors name into first, middle and last name
    first_name = first_and_last_name[0]
    last_name=''
    middle_name=''
    if len(first_and_last_name) == 2:
        last_name = first_and_last_name[1]
    if len(first_and_last_name) == 3:
        middle_name = first_and_last_name[1]
        last_name = first_and_last_name[2]
    
    results = api.GetSearch(raw_query="q=" +first_name+"%20"+middle_name+"%20"+last_name + "&count=100")
    cnt = len(results)              #Making a call for each actor on twitter
    for res in results:
        x = json.loads(str(res))
        actor_info={
            'name' : first_name + ' ' + last_name,
            'tags' : [],
            'count' : cnt
        }
        
        for item in x['hashtags']:
            actor_info['tags'].append(item['text'])
    actor_tags.append(actor_info)

In [66]:
for each in actors_abv_7[1300:1480]:
    each = each.replace("'",' ')
    first_and_last_name = each.split(" ")    #Splitting each actors name into first, middle and last name
    first_name = first_and_last_name[0]
    last_name=''
    middle_name=''
    if len(first_and_last_name) == 2:
        last_name = first_and_last_name[1]
    if len(first_and_last_name) == 3:
        middle_name = first_and_last_name[1]
        last_name = first_and_last_name[2]
    
    results = api.GetSearch(raw_query="q=" +first_name+"%20"+middle_name+"%20"+last_name + "&count=100")
    cnt = len(results)              #Making a call for each actor on twitter
    for res in results:
        x = json.loads(str(res))
        actor_info={
            'name' : first_name + ' ' + last_name,
            'tags' : [],
            'count' : cnt
        }
        
        for item in x['hashtags']:
            actor_info['tags'].append(item['text'])
    actor_tags.append(actor_info)

In [68]:
for each in actors_abv_7[1480:1660]:
    each = each.replace("'",' ')
    first_and_last_name = each.split(" ")    #Splitting each actors name into first, middle and last name
    first_name = first_and_last_name[0]
    last_name=''
    middle_name=''
    if len(first_and_last_name) == 2:
        last_name = first_and_last_name[1]
    if len(first_and_last_name) == 3:
        middle_name = first_and_last_name[1]
        last_name = first_and_last_name[2]
    
    results = api.GetSearch(raw_query="q=" +first_name+"%20"+middle_name+"%20"+last_name + "&count=100")
    cnt = len(results)              #Making a call for each actor on twitter
    for res in results:
        x = json.loads(str(res))
        actor_info={
            'name' : first_name + ' ' + last_name,
            'tags' : [],
            'count' : cnt
        }
        
        for item in x['hashtags']:
            actor_info['tags'].append(item['text'])
    actor_tags.append(actor_info)

In [72]:
for each in actors_abv_7[1660:1840]:
    each = each.replace("'",' ')
    first_and_last_name = each.split(" ")    #Splitting each actors name into first, middle and last name
    first_name = first_and_last_name[0]
    last_name=''
    middle_name=''
    if len(first_and_last_name) == 2:
        last_name = first_and_last_name[1]
    if len(first_and_last_name) == 3:
        middle_name = first_and_last_name[1]
        last_name = first_and_last_name[2]
    
    results = api.GetSearch(raw_query="q=" +first_name+"%20"+middle_name+"%20"+last_name + "&count=100")
    cnt = len(results)              #Making a call for each actor on twitter
    for res in results:
        x = json.loads(str(res))
        actor_info={
            'name' : first_name + ' ' + last_name,
            'tags' : [],
            'count' : cnt
        }
        
        for item in x['hashtags']:
            actor_info['tags'].append(item['text'])
    actor_tags.append(actor_info)

In [75]:
for each in actors_abv_7[1840:2020]:
    each = each.replace("'",' ')
    first_and_last_name = each.split(" ")    #Splitting each actors name into first, middle and last name
    first_name = first_and_last_name[0]
    last_name=''
    middle_name=''
    if len(first_and_last_name) == 2:
        last_name = first_and_last_name[1]
    if len(first_and_last_name) == 3:
        middle_name = first_and_last_name[1]
        last_name = first_and_last_name[2]
    
    results = api.GetSearch(raw_query="q=" +first_name+"%20"+middle_name+"%20"+last_name + "&count=100")
    cnt = len(results)              #Making a call for each actor on twitter
    for res in results:
        x = json.loads(str(res))
        actor_info={
            'name' : first_name + ' ' + last_name,
            'tags' : [],
            'count' : cnt
        }
        
        for item in x['hashtags']:
            actor_info['tags'].append(item['text'])
    actor_tags.append(actor_info)

In [78]:
for each in actors_abv_7[2020:2200]:
    each = each.replace("'",' ')
    first_and_last_name = each.split(" ")    #Splitting each actors name into first, middle and last name
    first_name = first_and_last_name[0]
    last_name=''
    middle_name=''
    if len(first_and_last_name) == 2:
        last_name = first_and_last_name[1]
    if len(first_and_last_name) == 3:
        middle_name = first_and_last_name[1]
        last_name = first_and_last_name[2]
    
    results = api.GetSearch(raw_query="q=" +first_name+"%20"+middle_name+"%20"+last_name + "&count=100")
    cnt = len(results)              #Making a call for each actor on twitter
    for res in results:
        x = json.loads(str(res))
        actor_info={
            'name' : first_name + ' ' + last_name,
            'tags' : [],
            'count' : cnt
        }
        
        for item in x['hashtags']:
            actor_info['tags'].append(item['text'])
    actor_tags.append(actor_info)

In [82]:
for each in actors_abv_7[2200:2380]:
    each = each.replace("'",' ')
    first_and_last_name = each.split(" ")    #Splitting each actors name into first, middle and last name
    first_name = first_and_last_name[0]
    last_name=''
    middle_name=''
    if len(first_and_last_name) == 2:
        last_name = first_and_last_name[1]
    if len(first_and_last_name) == 3:
        middle_name = first_and_last_name[1]
        last_name = first_and_last_name[2]
    
    results = api.GetSearch(raw_query="q=" +first_name+"%20"+middle_name+"%20"+last_name + "&count=100")
    cnt = len(results)              #Making a call for each actor on twitter
    for res in results:
        x = json.loads(str(res))
        actor_info={
            'name' : first_name + ' ' + last_name,
            'tags' : [],
            'count' : cnt
        }
        
        for item in x['hashtags']:
            actor_info['tags'].append(item['text'])
    actor_tags.append(actor_info)

In [85]:
for each in actors_abv_7[2400:2580]:
    each = each.replace("'",' ')
    first_and_last_name = each.split(" ")    #Splitting each actors name into first, middle and last name
    first_name = first_and_last_name[0]
    last_name=''
    middle_name=''
    if len(first_and_last_name) == 2:
        last_name = first_and_last_name[1]
    if len(first_and_last_name) == 3:
        middle_name = first_and_last_name[1]
        last_name = first_and_last_name[2]
    
    results = api.GetSearch(raw_query="q=" +first_name+"%20"+middle_name+"%20"+last_name + "&count=100")
    cnt = len(results)              #Making a call for each actor on twitter
    for res in results:
        x = json.loads(str(res))
        actor_info={
            'name' : first_name + ' ' + last_name,
            'tags' : [],
            'count' : cnt
        }
        
        for item in x['hashtags']:
            actor_info['tags'].append(item['text'])
    actor_tags.append(actor_info)

In [90]:
for each in actors_abv_7[2600:2780]:
    each = each.replace("'",' ')
    first_and_last_name = each.split(" ")    #Splitting each actors name into first, middle and last name
    first_name = first_and_last_name[0]
    last_name=''
    middle_name=''
    if len(first_and_last_name) == 2:
        last_name = first_and_last_name[1]
    if len(first_and_last_name) == 3:
        middle_name = first_and_last_name[1]
        last_name = first_and_last_name[2]
    
    results = api.GetSearch(raw_query="q=" +first_name+"%20"+middle_name+"%20"+last_name + "&count=100")
    cnt = len(results)              #Making a call for each actor on twitter
    for res in results:
        x = json.loads(str(res))
        actor_info={
            'name' : first_name + ' ' + last_name,
            'tags' : [],
            'count' : cnt
        }
        
        for item in x['hashtags']:
            actor_info['tags'].append(item['text'])
    actor_tags.append(actor_info)

In [93]:
for each in actors_abv_7[2800:2980]:
    each = each.replace("'",' ')
    first_and_last_name = each.split(" ")    #Splitting each actors name into first, middle and last name
    first_name = first_and_last_name[0]
    last_name=''
    middle_name=''
    if len(first_and_last_name) == 2:
        last_name = first_and_last_name[1]
    if len(first_and_last_name) == 3:
        middle_name = first_and_last_name[1]
        last_name = first_and_last_name[2]
    
    results = api.GetSearch(raw_query="q=" +first_name+"%20"+middle_name+"%20"+last_name + "&count=100")
    cnt = len(results)              #Making a call for each actor on twitter
    for res in results:
        x = json.loads(str(res))
        actor_info={
            'name' : first_name + ' ' + last_name,
            'tags' : [],
            'count' : cnt
        }
        
        for item in x['hashtags']:
            actor_info['tags'].append(item['text'])
    actor_tags.append(actor_info)

In [102]:
actor_tags

[{'count': 91,
  'name': 'Clifton Powell',
  'tags': ['CliftonPowell', 'OmarGooding']},
 {'count': 29, 'name': 'Ryan Neal', 'tags': []},
 {'count': 26, 'name': 'Veronica Cartwright', 'tags': []},
 {'count': 100, 'name': 'Bob Costas', 'tags': []},
 {'count': 100, 'name': 'Zach Braff', 'tags': []},
 {'count': 85, 'name': 'Paul Reiser', 'tags': []},
 {'count': 100, 'name': 'Slim Pickens', 'tags': []},
 {'count': 3, 'name': 'Stephen Trask', 'tags': []},
 {'count': 100, 'name': 'Diana Bracho', 'tags': []},
 {'count': 76, 'name': 'Diane Keaton', 'tags': []},
 {'count': 100, 'name': 'Luke Wilson', 'tags': []},
 {'count': 2, 'name': 'Ron Ben-Yishai', 'tags': ['Israel']},
 {'count': 3, 'name': 'Sally Dryer', 'tags': []},
 {'count': 3, 'name': 'Sally Dryer', 'tags': []},
 {'count': 33, 'name': 'Na Lynn', 'tags': []},
 {'count': 100, 'name': 'Claudia Cardinale', 'tags': ['AFP']},
 {'count': 3, 'name': 'Dolly Wells', 'tags': []},
 {'count': 20, 'name': 'Marilyn Lightstone', 'tags': []},
 {'count':

In [110]:
frames=[]    #Chunking to reduce runtime
for each in actor_tags[:500]:
    df = pd.DataFrame(each)
    frames.append(df)
    df_1=pd.concat(frames)

In [111]:
frames=[]
for each in actor_tags[500:1000]:
    df = pd.DataFrame(each)
    frames.append(df)
    df_2=pd.concat(frames)

In [112]:
frames=[]
for each in actor_tags[1000:1500]:
    df = pd.DataFrame(each)
    frames.append(df)
    df_3=pd.concat(frames)

In [113]:
frames=[]
for each in actor_tags[1500:2000]:
    df = pd.DataFrame(each)
    frames.append(df)
    df_4=pd.concat(frames)

In [114]:
frames=[]
for each in actor_tags[2000:2500]:
    df = pd.DataFrame(each)
    frames.append(df)
    df_5=pd.concat(frames)

In [115]:
frames=[]
for each in actor_tags[2500:]:
    df = pd.DataFrame(each)
    frames.append(df)
    df_6=pd.concat(frames)

In [129]:
tags_df = pd.concat([df_1,df_2,df_3,df_4,df_5,df_6])

In [234]:
tags_df.reset_index(drop=True,inplace=True)
tags_df[tags_df['count'] == 100]

count               name                             tags
3      100  Claudia Cardinale                              AFP
4      100         Dick Clark                      alwaysyoung
6      100      Albert Finney                               役名
7      100      Albert Finney                               映画
8      100      Albert Finney                               男優
9      100      Albert Finney                               英名
10     100      Albert Finney                            ドレッサー
11     100      Albert Finney                       アルバート・フィニー
12     100      Albert Finney                     AlbertFinney
13     100      Albert Finney                             オスカー
27     100        Thomas Jane                  FemaleSexuality
28     100        Thomas Jane                            women
29     100          Phil Fish                         salmonid
30     100          Phil Fish                             fish
31     100        Bette Davis                   BetteDavisEyes
34     100         Alan Arkin                           BAFTAs
41     100      Brian Blessed                          Elite11
42     100      Brian Blessed                              AO1
48     100        Simon Woods            PrideandPrejudice2005
49     100        Simon Woods                  ElizabethBennet
50     100        Simon Woods                        MrBingley
59     100      Brenda Strong                          bbnaija
64     100     Leonor Watling         AsesinatoEnLaUniversidad
66     100     Philippe Petit                 salonagriculture
67     100       Lillian Gish                         MissGish
70     100        Zoe Saldana                              KCA
71     100       Emily Watson                       ForOurCity
72     100     Richard Beymer                        twinpeaks
73     100     Richard Beymer                         BenHorne
76     100       Maggie Grace                         svnation
..     ...                ...                              ...
807    100    Kelly Macdonald                    TalDíaComoHoy
808    100    Kelly Macdonald                            23Feb
810    100       Vinnie Jones  awfullybritishtorturetechniques
812    100       Tupac Shakur                      UnsolvedUSA
814    100   Timothy Olyphant                 SantaClaritaDiet
815    100          Eric Bana                           Movies
816    100          Eric Bana                          Trailer
828    100       J.K. Simmons                      Counterpart
829    100         Eric Young                             Rams
830    100         Eric Young                             Rams
839    100       Helen Mirren                          Time100
841    100    Michiel Huisman                             Veep
842    100    Michiel Huisman                          Banshee
843    100    Michiel Huisman                 IrreplaceableYou
845    100         Evan Jones                       Nowplaying
849    100             Norma                       Tennisthink
852    100     Saurabh Shukla                             Raid
854    100        Arthur Hill                    TalDíaComoHoy
855    100        Arthur Hill                            23Feb
856    100      Marlon Wayans                            quote
857    100      Marlon Wayans                            image
858    100      Marlon Wayans                          believe
859    100      Marlon Wayans                           growth
860    100      Marlon Wayans                              win
862    100    Michael Emerson                              TBT
863    100    Michael Emerson                   MichaelEmerson
864    100    Michael Emerson                            Arrow
865    100    Michael Emerson                      CaydenJames
866    100       James Garner                             MUFC
870    100           Jude Law                    CaptainMarvel

[382 rows x 3 columns]

In [51]:
tags_df

count               name               tags
0       91     Clifton Powell      CliftonPowell
1       91     Clifton Powell        OmarGooding
2        2     Ron Ben-Yishai             Israel
3      100  Claudia Cardinale                AFP
4      100         Dick Clark        alwaysyoung
5        2    Roger Leenhardt         AndreBazin
6      100      Albert Finney                 役名
7      100      Albert Finney                 映画
8      100      Albert Finney                 男優
9      100      Albert Finney                 英名
10     100      Albert Finney              ドレッサー
11     100      Albert Finney         アルバート・フィニー
12     100      Albert Finney       AlbertFinney
13     100      Albert Finney               オスカー
14      37          Paul Kaye            RMPatPM
15      17        Gale Hansen            Vintage
16      17        Gale Hansen              Video
17      17        Gale Hansen             videos
18      17        Gale Hansen              SEALS
19      17        Gale Hansen               SEAL
20      17        Gale Hansen            GULFWAR
21      17        Gale Hansen            Vintage
22      17        Gale Hansen              Video
23      17        Gale Hansen             videos
24      17        Gale Hansen              SEALS
25      17        Gale Hansen               SEAL
26      17        Gale Hansen            GULFWAR
27     100        Thomas Jane    FemaleSexuality
28     100        Thomas Jane              women
29     100          Phil Fish           salmonid
..     ...                ...                ...
842    100    Michiel Huisman            Banshee
843    100    Michiel Huisman   IrreplaceableYou
844     77     Darren McGavin        metvkolchak
845    100         Evan Jones         Nowplaying
846     54     Maureen Lipman       Ambridgology
847     54     Maureen Lipman         thearchers
848     54     Maureen Lipman    academicarchers
849    100             Norma         Tennisthink
850      7  Brighton Sharbino            CelebVM
851      5      Rupert Davies            Maigret
852    100     Saurabh Shukla               Raid
853     86     Gerald McRaney   SantaClaritaDiet
854    100        Arthur Hill      TalDíaComoHoy
855    100        Arthur Hill              23Feb
856    100      Marlon Wayans              quote
857    100      Marlon Wayans              image
858    100      Marlon Wayans            believe
859    100      Marlon Wayans             growth
860    100      Marlon Wayans                win
861     25        Karin Viard            Jalouse
862    100    Michael Emerson                TBT
863    100    Michael Emerson     MichaelEmerson
864    100    Michael Emerson              Arrow
865    100    Michael Emerson        CaydenJames
866    100       James Garner               MUFC
867     20       Wendy Hiller  GeorgeBernardShaw
868     20       Wendy Hiller              Oscar
869     20       Wendy Hiller          Pygmalion
870    100           Jude Law      CaptainMarvel
871     81       Jonny Miller           Leggiamo

[872 rows x 3 columns]

In [47]:
users_sbst = users[100:300]
users_sbst

[{'id': 706214313967689728, 'name': '♠️ACE'},
 {'id': 3257783633, 'name': 'Richard Hallam, Bass'},
 {'id': 2964550103, 'name': 'Corinne B'},
 {'id': 227132957, 'name': 'JGreen Images'},
 {'id': 2550047628, 'name': 'Mike Lamb'},
 {'id': 4076402776, 'name': 'Marlene Danko'},
 {'id': 329429279, 'name': 'Professor Louie'},
 {'id': 931239607513092096, 'name': 'Les Vigorish'},
 {'id': 295599767, 'name': 'harold curran'},
 {'id': 507599300, 'name': 'Gun Club Music'},
 {'id': 2842614477, 'name': 'cotoncoton'},
 {'id': 202907955, 'name': 'Kristopher Gardham'},
 {'id': 812852177395847169, 'name': 'とし'},
 {'id': 838417063890812929, 'name': 'Bunwip 🐇'},
 {'id': 589283111, 'name': 'CorkScrew'},
 {'id': 920386732863848448, 'name': 'SundayMorningAnalog'},
 {'id': 3295902912, 'name': 'Coon-Doggie'},
 {'id': 348538041, 'name': 'na na na é o diabo!'},
 {'id': 2422517929, 'name': 'route 66'},
 {'id': 3082380272, 'name': 'Paul Gaudet'},
 {'id': 359624519, 'name': 'The Poetry E Train'},
 {'id': 68558349, '

In [48]:
users_sbst_id = []
for each in users_sbst:
    users_sbst_id.append(each['id'])
users_sbst_id = list(set(users_sbst_id))
users_sbst_id[18:33]

[17238552,
 227132957,
 248213023,
 2147631139,
 252409895,
 987220009,
 52595245,
 106812974,
 1035231792,
 856501132440555520,
 21496838,
 1565500998,
 503829063,
 488543818,
 4055129164]

In [72]:
followers_df_list=[]
for each in users_sbst_id[:15]:
    qry = api.GetFollowers(user_id= each ,skip_status=True)
    followers_user=[]
    for each in qry:
        x = json.loads(str(each))
        followers_info = {}
        followers_info['name'] = x['screen_name']
        followers_info['id'] = x['id']
        followers_user.append(followers_info)
    followers_df_list.append(followers_user)      #DONE

TwitterError: [{'message': 'Rate limit exceeded', 'code': 88}]

In [134]:
for each in users_sbst_id[4:15]:
    qry = api.GetFollowers(user_id= each ,skip_status=True)
    followers_user=[]
    for each in qry:
        x = json.loads(str(each))
        followers_info = {}
        followers_info['name'] = x['screen_name']
        followers_info['id'] = x['id']
        followers_user.append(followers_info)
    followers_df_list.append(followers_user)

TwitterError: [{'message': 'Rate limit exceeded', 'code': 88}]

In [65]:
friends_df_list=[]
for each in users_sbst_id[:4]:
    qry = api.GetFriends(user_id= each ,skip_status=True)
    friends_user=[]
    for each in qry:
        x = json.loads(str(each))
        friends_info = {}
        friends_info['name'] = x['screen_name']
        friends_info['id'] = x['id']
        friends_user.append(friends_info)
    friends_df_list.append(friends_user)    #DONE

TwitterError: [{'message': 'Rate limit exceeded', 'code': 88}]

In [83]:
for each in users_sbst_id[4:15]:
    qry = api.GetFriends(user_id= each ,skip_status=True)
    friends_user=[]
    for each in qry:
        x = json.loads(str(each))
        friends_info = {}
        friends_info['name'] = x['screen_name']
        friends_info['id'] = x['id']
        friends_user.append(friends_info)
    friends_df_list.append(friends_user)

TwitterError: [{'message': 'Rate limit exceeded', 'code': 88}]

In [135]:
count=0
for each in followers_df_list:
    count += len(each)
count
len(followers_df_list)

4

In [136]:
count=0
for each in friends_df_list:
    count += len(each)
count
len(friends_df_list)

4

In [318]:
%store tags_df

Stored 'tags_df' (DataFrame)


In [2]:
%store -r tags_df

In [87]:
%store followers_df_list

Stored 'followers_df_list' (list)


In [88]:
%store friends_df_list

Stored 'friends_df_list' (list)


In [95]:
frames=[]
for i,x in enumerate(followers_df_list[:]):
    for j,y in enumerate(x):
        df = pd.DataFrame(y, index=[i+j])
        frames.append(df)
followers_df = pd.concat(frames)

In [96]:
frames=[]
for i,x in enumerate(friends_df_list[:]):
    for j,y in enumerate(x):
        df = pd.DataFrame(y, index=[i+j])
        frames.append(df)
friends_df = pd.concat(frames)

In [105]:
friends_df.reset_index(drop=True,inplace=True)
friends_df.tail()

id            name
2407    93498250        XboxQwik
2408   701983068      GameGrumps
2409    13687132  engadgetgaming
2410  2305603543   TwitchEsports
2411    15838599    Capcom_Unity

In [106]:
followers_df.reset_index(drop=True,inplace=True)
followers_df.tail()

id             name
3145          3286042358     rachelaexton
3146  816985742148861952      FearedAlien
3147  810636691245592576  polygonfeatured
3148          4619670917       VRProGamer
3149  837739563909001220       PariseauHD

In [109]:
friends_df_list

[[{'id': 885399802489339904, 'name': 'Stay_Awayyyyy'},
  {'id': 875155600749076482, 'name': 'iAm_Falak'},
  {'id': 1107599149, 'name': 'dxyle'},
  {'id': 161421718, 'name': 'mdasif_2007'},
  {'id': 2244721284, 'name': 'ImShehrozAhmad'},
  {'id': 832868599228403712, 'name': 'AyeshaaMalik1'},
  {'id': 921681143736434693, 'name': 'AfzaKha16812032'},
  {'id': 854992681935699968, 'name': 'iamfarwaqamar'},
  {'id': 911134419284631552, 'name': 'Asli_nazia'},
  {'id': 886964992662269953, 'name': 'tayyab_AKQ'},
  {'id': 961688022646448128, 'name': 'its_yums_here'},
  {'id': 927946834588438530, 'name': 'Nimi_syed'},
  {'id': 2578021908, 'name': 'MisterShebi'},
  {'id': 954409016041537537, 'name': 'soul_hacker1234'},
  {'id': 873590847778041856, 'name': 'ullukidumm'},
  {'id': 903634580724490242, 'name': 'Raja_HassaN10'},
  {'id': 2862299244, 'name': 'iamnomanamjad'},
  {'id': 3446964253, 'name': 'idga_phuck'},
  {'id': 956446606123044865, 'name': 'hafsayyy__'},
  {'id': 933269110305296384, 'name

In [110]:
userid_friends_list=[]
for i,x in enumerate(friends_df_list):
    for y in x:
        userid_friends_list.append(users_sbst_id[i])

In [112]:
userid_followers_list =[]
for i,x in enumerate(followers_df_list):
    for y in x:
        userid_followers_list.append(users_sbst_id[i])

In [114]:
friends_df['user_id'] = pd.Series(userid_friends_list)
followers_df['user_id'] = pd.Series(userid_followers_list)

In [125]:
friends_df = friends_df[['user_id','name','id']]
friends_df.columns.values[2] = 'friend_id'
friends_df

user_id             name           friend_id
0     706214313967689728    Stay_Awayyyyy  885399802489339904
1     706214313967689728        iAm_Falak  875155600749076482
2     706214313967689728            dxyle          1107599149
3     706214313967689728      mdasif_2007           161421718
4     706214313967689728   ImShehrozAhmad          2244721284
5     706214313967689728    AyeshaaMalik1  832868599228403712
6     706214313967689728  AfzaKha16812032  921681143736434693
7     706214313967689728    iamfarwaqamar  854992681935699968
8     706214313967689728       Asli_nazia  911134419284631552
9     706214313967689728       tayyab_AKQ  886964992662269953
10    706214313967689728    its_yums_here  961688022646448128
11    706214313967689728        Nimi_syed  927946834588438530
12    706214313967689728      MisterShebi          2578021908
13    706214313967689728  soul_hacker1234  954409016041537537
14    706214313967689728       ullukidumm  873590847778041856
15    706214313967689728    Raja_HassaN10  903634580724490242
16    706214313967689728    iamnomanamjad          2862299244
17    706214313967689728       idga_phuck          3446964253
18    706214313967689728       hafsayyy__  956446606123044865
19    706214313967689728         no1sbaee  933269110305296384
20    706214313967689728       1qraAsghar          4226492254
21    706214313967689728      kamli_bajii  941549627735724032
22    706214313967689728      MarWaYYyYYy           712002823
23    706214313967689728    BrainWasher22  876813650090229760
24    706214313967689728       TadyBear12  953182156666343430
25    706214313967689728          mujji91           145591029
26    706214313967689728         Hk5Sanam  870920390666846208
27    706214313967689728    Thechampagne_  941317699774369793
28    706214313967689728        Khnsaa_88  726413126648881152
29    706214313967689728      Ghareeeb_AF  920617089559334912
...                  ...              ...                 ...
2382  838417063890812929       markiplier           517077573
2383  838417063890812929       ScufGaming           147558951
2384  838417063890812929           Bungie            26280712
2385  838417063890812929       Jam_sponge            19455489
2386  838417063890812929    MonsterGaming            60725654
2387  838417063890812929  NintendoAmerica             5162861
2388  838417063890812929  HIDEO_KOJIMA_EN           117652722
2389  838417063890812929  DavidVonderhaar            34068984
2390  838417063890812929          bioware            21158690
2391  838417063890812929  GearboxOfficial            16567106
2392  838417063890812929   therealcliffyb            48485771
2393  838417063890812929             yosp            35406821
2394  838417063890812929  PeanutButterGmr            59958288
2395  838417063890812929        earth2kvn          2855965952
2396  838417063890812929    geoffkeighley             9451052
2397  838417063890812929      Naughty_Dog            15222083
2398  838417063890812929   aarongreenberg            19595287
2399  838417063890812929      majornelson               15913
2400  838417063890812929        Hale_Fail           187218959
2401  838417063890812929           steveo           208711264
2402  838417063890812929         Mort3mer            18643769
2403  838417063890812929        NoMansSky          2232827756
2404  838417063890812929       GamesRadar             6240732
2405  838417063890812929          Polygon           454340464
2406  838417063890812929      ASTROGaming            18776017
2407  838417063890812929         XboxQwik            93498250
2408  838417063890812929       GameGrumps           701983068
2409  838417063890812929   engadgetgaming            13687132
2410  838417063890812929    TwitchEsports          2305603543
2411  838417063890812929     Capcom_Unity            15838599

[2412 rows x 3 columns]

In [130]:
followers_df  = followers_df[['user_id','name','id']]
followers_df.columns.values[2] = 'follower_id'
followers_df

user_id             name         follower_id
0     706214313967689728        userbenji          2454419568
1     706214313967689728    SlyTunesMusic           268065568
2     706214313967689728           ImJK93  883965037953470464
3     706214313967689728      albalawi_12           446586628
4     706214313967689728     Arsiikehtahy  903625325623599104
5     706214313967689728         luisray0           962467802
6     706214313967689728      Brienna_Boo           300992517
7     706214313967689728     zstylesfrost  919954839458549765
8     706214313967689728    Leonel_leito1           389134103
9     706214313967689728    Farrakhimtiaz           455419652
10    706214313967689728   DespicableMe__           302677041
11    706214313967689728     DIGTHEMODEST          3037669936
12    706214313967689728       CyrusKarim          1364425134
13    706214313967689728   emsjmusicdrive          2719163403
14    706214313967689728        suhogalxy          2727746375
15    706214313967689728      PrinceJAS0N  824523842307719169
16    706214313967689728      nkazimulomz          3750699436
17    706214313967689728     SherozWrites  925418259939315712
18    706214313967689728     ambikabaheti          3114344166
19    706214313967689728  M_FakharHussain          2331656072
20    706214313967689728   Fake_Tio_Tommo          2377501039
21    706214313967689728        itzKeezy_  946247154301046784
22    706214313967689728        Sammii444          2338626266
23    706214313967689728       marley1111            46582496
24    706214313967689728         xzoneone          1663629008
25    706214313967689728    bieberdepthed          2664377321
26    706214313967689728      SamirNasrat          1494347791
27    706214313967689728            dxyle          1107599149
28    706214313967689728      mdasif_2007           161421718
29    706214313967689728      noahfluegge  914428424978145281
...                  ...              ...                 ...
3120  838417063890812929       JustMyYosh  841997529118081028
3121  838417063890812929     LINKKNIGHT97  891945737868591104
3122  838417063890812929   Z_GamingKill10  890657052841955328
3123  838417063890812929     grindmodegzy  731479231004327936
3124  838417063890812929    pixelthesloth  885891084852842497
3125  838417063890812929   contempo_games  894643075498221568
3126  838417063890812929              XGN          1135921740
3127  838417063890812929      LunnikaHoro           333469842
3128  838417063890812929  stevens70847794  894299744159379457
3129  838417063890812929    Galfano_No_10  881721476176596992
3130  838417063890812929  ManchesterisRED          3449226256
3131  838417063890812929      RedorDead07           876182094
3132  838417063890812929       petit_soul           350023195
3133  838417063890812929    PIXELHUNTER__  854968930049380354
3134  838417063890812929    maiya_mccrary  843912689651015680
3135  838417063890812929     GAMEWestwood            64541166
3136  838417063890812929       CrocOclock            63987129
3137  838417063890812929       SickCritic  741397342247493632
3138  838417063890812929    mari_tabassum  889063913517596672
3139  838417063890812929   500greatalbums           100736229
3140  838417063890812929  AndrewW77658969  891267024374165505
3141  838417063890812929     wolves_david  814711053476954112
3142  838417063890812929  missywithcheese          2401828014
3143  838417063890812929     xWolves_adam  814720816021532672
3144  838417063890812929         Cupgamez          3840089534
3145  838417063890812929     rachelaexton          3286042358
3146  838417063890812929      FearedAlien  816985742148861952
3147  838417063890812929  polygonfeatured  810636691245592576
3148  838417063890812929       VRProGamer          4619670917
3149  838417063890812929       PariseauHD  837739563909001220

[3150 rows x 3 columns]

In [133]:
import sqlite3
conn = sqlite3.connect('TMDB_Database')

In [137]:
movies_df.to_sql('movies',con=conn,index=False, if_exists='replace')
countries_df.to_sql('shoot_venue', con=conn, index=False, if_exists='replace')

In [138]:
prod_companies_df.to_sql('production_companies', con=conn, index=False, if_exists='replace')
cast_df.to_sql('cast', con=conn, index=False, if_exists='replace')
genre_df.to_sql('genre', con=conn, index=False, if_exists='replace')

In [139]:
tags_df.to_sql('tags', con=conn, index=False, if_exists='replace')
friends_df.to_sql('friends', con=conn, index=False, if_exists='replace')
followers_df.to_sql('followers', con=conn, index=False, if_exists='replace')

In [140]:
def run_query(query):
    return pd.read_sql(query,conn)

In [141]:
run_query('SELECT name FROM sqlite_master WHERE type = "table"')

name
0                movies
1           shoot_venue
2  production_companies
3                  cast
4                 genre
5                  tags
6               friends
7             followers

<img src='TMDB_schema.png'>

##  1. What are the tags associated with each actor ?

In [153]:
run_query('SELECT name, tags FROM tags ORDER BY name ')

name                tags
0         Abigail Breslin          DanceRadio
1            Adil Hussain          RIPSridevi
2            Adolph Davis            Antiques
3            Adolph Davis        Collectables
4            Adolph Davis             Auction
5            Adolph Davis             Vintage
6            Adolph Davis              Silver
7            Adolph Davis            Tapestry
8            Adolph Davis               Purse
9            Adolph Davis                 Bag
10           Adolph Davis            Antiques
11           Adolph Davis        Collectables
12           Adolph Davis             Auction
13           Adolph Davis             Vintage
14           Adolph Davis              Silver
15           Adolph Davis            Tapestry
16           Adolph Davis               Purse
17           Adolph Davis                 Bag
18   Aitana Sánchez-Gijón    ConcejalaCultura
19   Aitana Sánchez-Gijón          AnaRedondo
20              Al Pacino            HBOFilms
21              Al Pacino          PaternoHBO
22             Alan Arkin              BAFTAs
23              Alan Bean                nasa
24              Alan Bean               space
25              Alan Bean                nasa
26              Alan Bean               space
27          Albert Finney                  役名
28          Albert Finney                  映画
29          Albert Finney                  男優
..                    ...                 ...
842          Wendy Hiller   GeorgeBernardShaw
843          Wendy Hiller               Oscar
844          Wendy Hiller           Pygmalion
845         William Black             CineMon
846     William Sylvester  memorablemanitoban
847           Wood Harris           midufinga
848           Wood Harris                news
849           Wood Harris             thewire
850           Wood Harris          paidinfull
851         Xavier Samuel             RiotABC
852         Xavier Samuel             RiotABC
853             Yo Oizumi                佐々木希
854             Yo Oizumi               秋田ある記
855           Yohana Cobo              Volver
856              Yuen Wah           อุ่นไอรัก
857              Yuen Wah            มุมมหาชน
858              Yuen Wah           nightview
859              Yuen Wah                 bkk
860              Yuen Wah            thailand
861              Yuen Wah              garden
862           Yul Vazquez       MidnightTexas
863      Zachary Knighton    SantaClaritaDiet
864          Zachary Levi              Shazam
865          Zachary Levi              Shazam
866            Zainab Jah                 TBT
867            Zainab Jah                 TBT
868           Zhang Xinyi               SNH48
869           Zhang Xinyi               BEJ48
870           Zoe Saldana                 KCA
871           Zoë Kravitz     Entretenimiento

[872 rows x 2 columns]

## 2. What social media users are like other social media users in your domain?

In [158]:
run_query('SELECT friends.user_id FROM friends INNER JOIN followers ON friends.user_id = followers.user_id GROUP BY follower_id LIMIT 4')

user_id
0  812852177395847169
1  838417063890812929
2  838417063890812929
3  838417063890812929

Converting user_id to twitter name we get user_id 812852177395847169 => @toshi50redcat & 838417063890812929 => @BunwipBunwi
suggesting these two users are like each other

## 3. Which actors are popular in your domain?

In [162]:
run_query('SELECT name Actor_name, count Tweet_count FROM tags GROUP BY name ORDER BY count DESC LIMIT 10')

Actor_name  Tweet_count
0  Abigail Breslin          100
1     Adil Hussain          100
2        Al Pacino          100
3       Alan Arkin          100
4        Alan Bean          100
5    Albert Finney          100
6     Alexis Smith          100
7    Alfre Woodard          100
8    Amber Tamblyn          100
9      André Giant          100

## 4. Which actors are trending in your domain?

In [164]:
run_query('SELECT name Trending_actors, count Tweet_count FROM tags WHERE COUNT = 100 GROUP BY name')

Trending_actors  Tweet_count
0         Abigail Breslin          100
1            Adil Hussain          100
2               Al Pacino          100
3              Alan Arkin          100
4               Alan Bean          100
5           Albert Finney          100
6            Alexis Smith          100
7           Alfre Woodard          100
8           Amber Tamblyn          100
9             André Giant          100
10            Andy García          100
11        AnnaSophia Robb          100
12         Annette Bening          100
13            Anupam Kher          100
14            Arthur Hill          100
15            Belén Rueda          100
16   Benedict Cumberbatch          100
17            Bette Davis          100
18              Bill Duke          100
19         Billy Williams          100
20           Blake Cooper          100
21           Blake Ritson          100
22          Brenda Strong          100
23          Brian Blessed          100
24              Brian Cox          100
25             Brian Howe          100
26      Bruce Springsteen          100
27           Bruce Willis          100
28         Carey Mulligan          100
29          Carlo Verdone          100
..                    ...          ...
178          Sarah Parker          100
179        Saurabh Shukla          100
180        Sebastian Stan          100
181          Selena Gomez          100
182           Simon Woods          100
183          Steve Burton          100
184          Steve Carell          100
185          Steve Coogan          100
186         Steve McQueen          100
187            Taye Diggs          100
188         Taylor Kitsch          100
189        Teagan Presley          100
190             Ten After          100
191             Teri Garr          100
192        Thandie Newton          100
193           Thomas Jane          100
194      Timothy Olyphant          100
195              Tom Ward          100
196           Tommy Jones          100
197         Toni Collette          100
198        Toshirō Mifune          100
199          Tupac Shakur          100
200      Vanessa Martinez          100
201          Vinnie Jones          100
202           Walt Disney          100
203         William Black          100
204           Wood Harris          100
205         Xavier Samuel          100
206           Zoe Saldana          100
207           Zoë Kravitz          100

[208 rows x 2 columns]